### PR TITLE
Jetpack Onboarding: Require JPC and introduce intro & success for Business Address

### LINKED ISSUE #22528: Jetpack Onboarding: Show an intro screen on Business Address step
Instead of showing the business address form immediately, let's show this step first. This gives us an opportunity to show that the business address and map widget are part of Jetpack and require a connection.

The second paragraph in the description should be conditionally shown only if they skipped step 4 and are not connected yet.

Clicking the call to action button leads to the actual form screen, or to the Jetpack Connect flow if the user hasn't connected yet (see #22529 for that).

For more information see p6TEKc-1Rd-p2

### LINKED ISSUE #22529: Jetpack Onboarding: Lead to Jetpack Connect flow if intending to add business address
#22528 introduces a new screen with a CTA that will lead to the business address form. If the user's not connected yet, instead of leading to the business address form, let's lead to the Jetpack Connect flow, and after connecting, redirect to the form. If the user is already connected, the form will be shown immediately.

Note that we'll need to preserve the JPO wizard navigation back / skip links at the bottom while in the JPC flow.

For more information see p6TEKc-1Rd-p2

### LINKED ISSUE #22531: Jetpack Onboarding: Introduce a success screen for Business Address step
If the users have connected successfully and inserted a business address, we will display this screen instead of the intro screen/connection flow/business address form. If they haven't inserted an address they should see the form, and if they haven't connected, we should lead them to the JPC flow if they try to reach the business address step again.

It's worth to mention that this is the point where we should move the actual business address insertion to, because it should happen only after the user has connected.

For more information see p6TEKc-1Rd-p2

### PR DESCRIPTION
**Pull request by tyxla**
### Purpose

This PR updates the business address form step to add an intro screen (see #22528), require Jetpack Connect (see #22529) and introduces a success screen (see #22531), as suggested in p6TEKc-1Rd-p2. Uses some of the logic we applied in #22621.

Fixes #22528
Fixes #22529 
Fixes #22531 

### How it works

* The user loads the business address step and sees the intro screen with a CTA.
* Clicking the button will behave differently, depending on whether the site is currently connected or not:
  * If connected, will display the business address form.
  * If not connected, will redirect to Jetpack Connect and will start the connection flow. After finishing connection, will redirect back to same step, but will show the business address form.
* After submitting the business address form successfully, the success screen is presented to the user.
* The success screen will be presented to the user in the event they already have input their business address.

### Notes

**Note 1:** right now, after connecting, Jetpack Connect will not redirect to the business address step. This requires additional changes to Jetpack Connect that we'll handle in a separate PR. This will work by passing a URL parameter to JPC so it knows where to redirect to. That URL will consist of the JPO business address page URL with a `action=add_business_address` parameter, which will update the flow to directly show the form, following the user's original intent to click the "Add a business address" button.

**Note 2:** right now the PR will display the "Skip" links in the footer, contrary to the design mockups - this will be addressed separately, see #22641.

**Note 3:** now it seems to me that the error states that we display initially when showing the business address form are a little counter-intuitive. It would probably make sense for them to appear only when the user has tried to submit the form with 1 or more empty required fields. What do you think @rickybanister @joanrho?

### Preview:

**Business Address step - intro screen**

**Business Address step - form initially displayed**

**Business Address step - form with filled data**

**Business Address step - success screen**

### Testing
* Checkout this branch on your local Calypso.
* Start with a fresh Jetpack site with the latest **Jetpack master**
* Head to `/wp-admin/admin.php?page=jetpack&action=onboard&calypso_env=development` on the Jetpack site to start the onboarding flow.
* In the site type step, select **Business**.
* Skip steps until you reach the Business Address step.
* Verify the step appears properly as shown on the screenshots.
* Click the **Add a business address** button.
* You will be redirected to Jetpack Connect and connection process will start automatically.
* After connecting, it will redirect you to the default redirection place of JPC (this is expected, see the note above).
* Visit `http://calypso.localhost:3000/jetpack/start/business-address/YOURJETPACKSITE.COM?action=add_business_address` to simulate coming back from Jetpack Connect after a successful connection.
* Verify you can see the business address form.
* Input some data in all of the fields and submit the form.
* Verify you can see the success screen (as shown on the screenshot)
* Verify the contact info widget was inserted in the main sidebar on your remote site.
* Refresh the page and verify you can still see the success screen.
* Verify the **Continue** button leads to the next step.
* Go back to other steps and visit the Business Address step; verify you can still see the success screen.

### CODE REVIEW

File: client/jetpack-onboarding/steps/business-address.jsx [hunk 1]
```diff
@@ -71,52 +126,75 @@ class JetpackOnboardingBusinessAddressStep extends React.PureComponent {
...
-				) }
-				<br />
-				{ translate(
-					'You can add a map based on this information and change where this widget is located later on.'
-				) }
+				{ this.renderHeader() }
+
+				<TileGrid>
+					<Tile
+						buttonLabel={ translate( 'Add a business address' ) }
```
Comment (a8ci18n) [inline client/jetpack-onboarding/steps/business-address.jsx]: Hi! I've found a possible matching string that has already been translated **19** times:
`translate( 'Add a business address.' )` *ES Score: 16*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/jpo-ba-step-jpc-intro-and-success)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by joanrho** [comment]
> Verify the contact info widget was inserted in the main sidebar on your remote site.

@tyxla, regarding this step ^, I am able to see the Contact Form added in the /wp-admin/widgets.php view, but I don't see it's added in Customizer>Widgets>Blog Sidebar. Screenshots: https://cloudup.com/cq9XosnTl9L.

Other than that inconsistency, all else works as expected.

**Comment by tyxla** [comment]
> The logic that's in charge of displaying either the action tile or the form (or the success screen, for that matter) is getting a bit crowded and might deserve some componentization.

Indeed, we could spend some time reducing some duplication after we land all the essential bits of functionality.

**Comment by tyxla** [comment]
> regarding this step ^, I am able to see the Contact Form added in the /wp-admin/widgets.php view, but I don't see it's added in Customizer>Widgets>Blog Sidebar. Screenshots: https://cloudup.com/cq9XosnTl9L.

@joanrho, I am unable to reproduce this one. Are you sure you're looking at the same site? Because on your screenshots I can see a huge discrepancy between the list of widgets in /wp-admin/widgets and in the customizer - it's not just that the contact widget is missing, but the entire set of widgets is completely different.

**Comment by tyxla** [comment]
Merging this one, thank you folks 👍

**Comment by tyxla** [comment]
Also opened #22842 for discussing note 3 separately.

**Pull request merged**